glyphs-reader/src/font.rs

```diff
@@ -1502,17 +1502,54 @@ fn default_master_idx(raw_font: &RawFont) -> usize {
                 .enumerate()
                 .find(|(_, master)| master.id == origin)
                 .map(|(idx, _)| idx)
-                .unwrap_or(0)
         })
-        // TODO: implement searching for "a base style shared between all masters" as glyphsLib does
-        // Still nothing? - just look for one called Regular
-        .or_else(|| {
-            raw_font
-                .font_master
-                .iter()
-                .position(|m| matches!(m.name.as_deref(), Some("Regular")))
+    {
+        return master_idx;
+    }
+    // Look for a master whose style is shared by all masters, defaulting to 'Regular'
+    let base_style = find_base_style(&raw_font.font_master).unwrap_or("Regular".to_string());
+    if let Some(master_idx) = raw_font
+        .font_master
+        .iter()
+        .position(|m| m.name.as_ref() == Some(&base_style))
+    {
+        return master_idx;
+    }
+    // Second try, this time omit 'Regular' from the master names
+    raw_font
+        .font_master
+        .iter()
+        .position(|m| {
+            m.name
+                .as_ref()
+                .map(|name| {
+                    name.split_whitespace()
+                        .filter(|n| *n != "Regular")
+                        .collect::<Vec<_>>()
+                        .join(" ")
+                })
+                .as_ref()
+                == Some(&base_style)
         })
-        .unwrap_or_default()
+        .unwrap_or_default() // just pick the first
+}
+
+// Direct port of glyphsLib's find_base_style:
+// https://github.com/googlefonts/glyphsLib/blob/9d5828d874110c42dfc5f542db8eb84f88641eb5/Lib/glyphsLib/builder/axes.py#L652-L663
+fn find_base_style(masters: &[RawFontMaster]) -> Option<String> {
+    let master_names: Vec<_> = masters.iter().filter_map(|m| m.name.as_deref()).collect();
+    if master_names.is_empty() {
+        return None;
+    }
+
+    let mut base_style: Vec<_> = master_names[0].split_whitespace().collect();
+
+    for master_name in master_names {
+        let style: Vec<_> = master_name.split_whitespace().collect();
+        base_style.retain(|word| style.contains(word));
+    }
+
+    Some(base_style.join(" ")).filter(|s| !s.is_empty())
```

Comment:
filter should be unnecessary, `split_whitespace` doesn't return empty strings.

(also worth noting[1] just to be academic that this function is lossy if the input contained whitespace that weren't a space)

1. okay it's extremely debatable that this is 'worth noting'

Reply:
thanks for the comment. it should no longer apply though, given that #1001 isn't doing that any more